x-pack/osquerybeat/cmd/root.go

```diff
@@ -74,14 +75,30 @@ func genVerifyCmd(_ instance.Settings) *cobra.Command {
 func osquerybeatCfg(rawIn *proto.UnitExpectedConfig, agentInfo *client.AgentInfo) ([]*reload.ConfigWithMeta, error) {
 	// Convert to streams, osquerybeat doesn't use streams
 	streams := make([]*proto.Stream, 1)
+
+	// Enforce the datastream dataset and type because the libbeat call to CreateInputsFromStreams
+	// provides it's own defaults that are breaking the osquery with logstash
+	// The target datastream for the publisher is expected to be logs-osquery_manager.result-<namespace>
+	// while the libebeat management.CreateInputsFromStreams defaults to osquery-generic-default
+	var datastream *proto.DataStream
+	if rawIn.GetDataStream() != nil {
+		// Copy by value and modify dataset and type
+		ds := *rawIn.GetDataStream()
+		ds.Dataset = config.DefaultDataset
+		ds.Type = config.DefaultType
+		datastream = &ds
```

Comment:
This works, but if the dataset or type ever becomes configurable it will break. That's probably unlikely though.